@W-9016142 - Lwc Quick Action Recipes

asrinivas-a opened the pull request:
### What does this PR do?

This PR has recipes/examples of how to create and use LWC Quick Actions that was GA'd in 232

## The PR fulfills these requirements:

[Y] Tests for the proposed changes have been added/updated.
[Y] Code linting and formatting was performed.

KamilSzostak commented on force-app/main/default/lwc/showCustomUiScreenAction/showCustomUiScreenAction.css:
```diff
@@ -0,0 +1,539 @@
+/**
+ * 98.css
```
We may need to check with legal if we're allowed to use this CSS styles. It might be safe to build our own custom UI.

adityanaag3 commented:
@asrinivas-a Thank you for submitting this PR that includes the much-awaited Quick Actions. A couple of things that I noticed:
1. Currently there is no way from the UI to execute these quick actions. Can you please add the actions to the contact detail page and commit those changes, please?
2. `editRecordScreenAction` and `recordContextScreen` action are doing the same thing (in terms of the feature shown in recipe) . `recordContextScreen` can be removed in my opinion.
3. Can you please update `editRecordScreenAction` to use Contact instead of Account. All the recipes use Contacts. This way you can add all the Quick Actions on the Contact Detail page.
3. Can we modify either `dispatchEventHeadlessAction` or `navigateToRecordHeadlessAction` to include the `recordId`. That way, we will have 1 recipe for ScreenAction, 1 recipe for headless action with `recordId`, and one without `recordId`. (Also - I don't see the ability to create a Global Action using LWC)
4. `showCustomUiScreenAction` doesn't seem to add much value as a "recipe for Quick Actions", and can be removed in my opinion. We encourage SLDS and Base components as much as possible.

adityanaag3 reviewed (changes requested):
Please refer to https://github.com/trailheadapps/lwc-recipes/pull/522#issuecomment-871633483

asrinivas-a commented:
> @asrinivas-a Thank you for submitting this PR that includes the much-awaited Quick Actions. A couple of things that I noticed:
> 
> 1. Currently there is no way to execute these quick actions from the UI. Can you please add the actions to the contact detail page and commit those changes, please?
> 2. `editRecordScreenAction` and `recordContextScreen` action are doing the same thing (in terms of the feature shown in recipe) . `recordContextScreen` can be removed in my opinion.
> 3. Can you please update `editRecordScreenAction` to use Contact instead of Account. All the recipes use Contacts. This way you can add all the Quick Actions on the Contact Detail page.
> 4. Can we modify either `dispatchEventHeadlessAction` or `navigateToRecordHeadlessAction` to include the `recordId`. That way, we will have 1 recipe for ScreenAction, 1 recipe for headless action with `recordId`, and one without `recordId`. (Also - I don't see the ability to create a Global Action using LWC)
> 5. `showCustomUiScreenAction` doesn't seem to add much value as a "recipe for Quick Actions", and can be removed in my opinion. We encourage SLDS and Base components as much as possible.



> @asrinivas-a Thank you for submitting this PR that includes the much-awaited Quick Actions. A couple of things that I noticed:
> 
> 1. Currently there is no way to execute these quick actions from the UI. Can you please add the actions to the contact detail page and commit those changes, please?
> 2. `editRecordScreenAction` and `recordContextScreen` action are doing the same thing (in terms of the feature shown in recipe) . `recordContextScreen` can be removed in my opinion.
> 3. Can you please update `editRecordScreenAction` to use Contact instead of Account. All the recipes use Contacts. This way you can add all the Quick Actions on the Contact Detail page.
> 4. Can we modify either `dispatchEventHeadlessAction` or `navigateToRecordHeadlessAction` to include the `recordId`. That way, we will have 1 recipe for ScreenAction, 1 recipe for headless action with `recordId`, and one without `recordId`. (Also - I don't see the ability to create a Global Action using LWC)
> 5. `showCustomUiScreenAction` doesn't seem to add much value as a "recipe for Quick Actions", and can be removed in my opinion. We encourage SLDS and Base components as much as possible.

@adityanaag3  Addressed the comments . Can you check?

satyasekharcvb reviewed (commented):
It looks like the flexipage is created from the org that is enabled with Dynamic Interactions.

adityanaag3 commented:
@asrinivas-a 
1. As Satya mentioned, please disable dynamic interactions on the flexipage (we only rollout GA features to the recipes app).
2. Actions (created on the object) seem to be missing from the commit.

asrinivas-a commented:
> @asrinivas-a
> 
> 1. As Satya mentioned, please disable dynamic interactions on the flexipage (we only rollout GA features to the recipes app).
> 2. Actions (created on the object) seem to be missing from the commit.

1. Where have i enabled Dynamic Interactions? Did you mean Dynamic Actions feature? The feature is GA and if we want to add LWC Quick Actions via flexipage this is the only way to do it.
2. With the above Flexipage, external developers need to create the action using the deployed LWC Component

> @asrinivas-a
> 
> 1. As Satya mentioned, please disable dynamic interactions on the flexipage (we only rollout GA features to the recipes app).
> 2. Actions (created on the object) seem to be missing from the commit.

@adityanaag3 @satyasekharcvb 
1. Where have i enabled Dynamic Interactions? Did you mean dynamic actions? This is the only way to add actions on a Flexipage.

2. The other actions are out of the box actions coming from the Platform so we dont need to create actions on the object. ONLY for LWC Quick Actions the developer needs to create the action with name 'EditRecordAction' in setup page using the deployed LWC Component.

adityanaag3 commented:
Thanks @asrinivas-a - Looks good. However, we are waiting for a new version of `@salesforce/sfdx-lwc-jest` that fixes a few issues, and also there is a problem with the CI (which we will fix shortly). Once both of these are addressed from our end, I will merge the PR. No action needed from you for now.

KamilSzostak commented:
@adityanaag3, have both those issues been addressed? Can we merge this PR?
